tests/unit_test/channels/gupshup_test.py

```
@@ -0,0 +1,1030 @@
+    async def test_send_action_async_else_branch_json_parse_fails_falls_back_to_text(self, gupshup_client):
+        """Non-200/202 response where .json() raises — should fall back to .text()."""
+        from aioresponses import aioresponses
+        url = "https://partner.gupshup.io/partner/app/app_001/msg"
+        with aioresponses() as m:
+            m.post(url, body=b"internal server error", status=500,
+                   headers={"Content-Type": "text/plain"})
+            m.post(url, body=b"internal server error", status=500,
+                   headers={"Content-Type": "text/plain"})
+            m.post(url, body=b"internal server error", status=500,
+                   headers={"Content-Type": "text/plain"})
+            ok, status, resp = await gupshup_client.send_action_async(
```

Ruff (0.15.21)

[warning] Unpacked variable `resp` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)